src/main/kotlin/org/jetbrains/research/testspark/tools/TestCompilerFactory.kt

```diff
@@ -3,20 +3,31 @@ package org.jetbrains.research.testspark.tools
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.roots.ProjectRootManager
 import org.jetbrains.research.testspark.core.data.JUnitVersion
+import org.jetbrains.research.testspark.core.test.SupportedLanguage
 import org.jetbrains.research.testspark.core.test.TestCompiler
+import org.jetbrains.research.testspark.core.test.java.JavaTestCompiler
+import org.jetbrains.research.testspark.core.test.kotlin.KotlinTestCompiler
 
 class TestCompilerFactory {
     companion object {
-        fun createJavacTestCompiler(
+        fun createTestCompiler(
             project: Project,
             junitVersion: JUnitVersion,
```

Comment:
We can also turn this factory into a factory method within the `TestCompiler` interface.